core/templates/dev/head/pages/exploration_player/ExtractImageFilenamesFromStateService.js

```diff
@@ -108,32 +108,112 @@ oppia.factory('ExtractImageFilenamesFromStateService', [
     };
 
     /**
-     * Extracts the filenames from the filepath-value attribute of the
+     * Extracts the filepath object from the filepath-value attribute of the
      * oppia-noninteractive-image tags in the strHtml(given string).
-     * @param {string} strHtml - The string from which the filenames of images
-     *                           should be extracted.
+     * @param {string} strHtml - The string from which the object of
+     *                           filepath should be extracted.
      */
     var _extractFilepathValueFromOppiaNonInteractiveImageTag = function(
```

Comment:
Now the _extractFilepathValueFromOppiaNonInteractiveImageTag function returns an array of the filepath objects.

And we have separate methods ---
_getImageFilenamesFromFilepathValue --- to get the filenames in the state using this method. (use the name field of the filepath object)

_getImageDimensionsFromFilepathValue --- to get the dimensions of the images in the state using this method( using the width and the height attribute of the filepath object)